test/eventmesh.LockManager.spec.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+const Promise = require('bluebird');
+const manager = require('../eventmesh').lockManager;
+const CONST = require('../eventmesh/constants');
+
+const {
+  Etcd3
+} = require('etcd3');
+
+describe('eventmesh', () => {
+  describe('LockManager', () => {
+    let sandbox, valueStub, stringStub, jsonStub, putStub, getStub, lockStub;
+    before(() => {
+      sandbox = sinon.sandbox.create();
+      valueStub = sandbox.stub();
+      stringStub = sandbox.stub();
+      jsonStub = sandbox.stub();
+      putStub = sandbox.stub(Etcd3.prototype, 'put', () => {
+        return {
+          value: (val) => Promise.resolve(valueStub(val))
+        };
+      });
+      getStub = sandbox.stub(Etcd3.prototype, 'get', () => {
+        return {
+          json: () => Promise.resolve(jsonStub())
+        };
+      });
+      lockStub = sandbox.stub(Etcd3.prototype, 'lock', () => {
+        return {
+          ttl: () => {
+            return {
+              acquire: () => Promise.resolve(stringStub())
+            };
+          },
+          release: () => Promise.resolve(stringStub())
+        };
+      });
+    });
+
+    afterEach(function () {
+      valueStub.reset();
+      putStub.reset();
+      getStub.reset();
+      jsonStub.reset();
+      stringStub.reset();
+    });
+    after(function () {
+      sandbox.restore();
+    });
+
+    describe('#isWriteLocked', () => {
+      it('should return false in case the resource has no lock', () => {
+        return manager.isWriteLocked('fakeResource')
+          .then(() => {
+            /* jshint expr: true */
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+      it('should return true for a write lock.', () => {
+        const writeLockResp = {
+          'count': 1,
+          'operationType': 'WRITE'
+        };
+        jsonStub.onCall(0).returns(writeLockResp);
+        return manager.isWriteLocked('fakeResource')
+          .then(result => {
+            /* jshint expr: true */
+            expect(result).to.eql(true);
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+      it('should return false for a read lock.', () => {
+        const readLockResp = {
+          'count': 1,
+          'operationType': 'READ'
+        };
+        jsonStub.onCall(0).returns(readLockResp);
+        return manager.isWriteLocked('fakeResource')
+          .then(result => {
+            /* jshint expr: true */
+            expect(result).to.eql(false);
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+      it('should return false for a no lock.', () => {
+        const noLockResp = {
+          'count': 0,
+          'operationType': ''
+        };
```

Comment:
What will be the response of client.get(key).json() if key is not there? How will it know this schema of response? Shouldn't it be null or undefined or something similar?

Reply:
Yes, the first test of this block basically tests that where I do not set any return value for the same which would return undefined.